### PR TITLE
chore(build): add workflow for building images 

### PR DESCRIPTION
**Pull request by sachk**
Had to turn off image optimization in next.config.js for next export to work

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Docker
+on:
+  push:
+    branches:
+      - "master"
```
Comment (lhjt) [inline .github/workflows/docker.yml]: Wrong branch

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          password: ${{ secrets.GH_TOKEN }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          push: ${{ github.event_name != 'pull_request' }}
+          platforms: linux/amd64
+          file: Dockerfile
+          tags: |
+            ghcr.io/csesoc/tech-spire-2022-${{ matrix.component }}:${{ github.sha }}
+            ghcr.io/csesoc/tech-spire-2022-${{ matrix.component }}:latest
```
Comment (lhjt) [inline .github/workflows/docker.yml]: Cannot use `matrix` here as a matrix strategy is not defined

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1 @@
+.github/ @csesoc/technical
```
Comment (lhjt) [inline CODEOWNERS:1]: Add Dockerfile

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM node:18-alpine as builder
+
+WORKDIR /app
+
+COPY . .
+
+RUN npm install && \
+	npx next build && \
+	npx next export
```
Comment (lhjt) [inline Dockerfile]: Can add another line to `package.json` called `build:ci` or something that just runs `next export`

Comment (lhjt) [reply Dockerfile]: And then use that here

File: nginx.conf [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    server {
+        listen 80;
+        server_name techspire.csesoc.unsw.edu.au;
+
+        root /usr/share/nginx/html/;
+        index /index.html;
+
+        location / {
+            include  /etc/nginx/mime.types;
+            try_files $uri /index.html;
```
Comment (lhjt) [inline nginx.conf]: I believe this line isn't necessary as it produces multiple HTML files at the appropriate routes (I think)

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,17 @@
+FROM node:18-alpine as builder
```
Comment (lhjt) [inline Dockerfile]: Pin the version please

**Review by lhjt — CHANGES_REQUESTED** [review]

File: nginx.conf (hunk 5 above)
Comment (sachk) [reply nginx.conf]: this is true

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,17 @@
+FROM node:18.10.0-alpine as builder
+
+WORKDIR /app
+
+COPY . .
+
+RUN npm install && \
```
Comment (lhjt) [inline Dockerfile]: Optimise this layer please :3

**Review by lhjt — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 7 above)
Comment (sachk) [reply Dockerfile]: > Optimise this layer please :3

wdym by this? it's also a build layer.

Comment (sachk) [reply Dockerfile]: oh do you mean splitting up package.json for faster builds?

Comment (lhjt) [reply Dockerfile]: I meant for caching - copy the lockfiles and package.json over in a separate layer and run npm install. The actual nginx layer is what is exported so it doesn't matter if the dep install layer is separate instead.

**Review by lhjt — APPROVED** [review]

**Pull request merged**